### ISSUE TITLE
📍 File picker dialog improvements

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/server#38926]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**
The current file picker is very basic at the moment – after talking with @jospoortvliet about it we identified some improvements:

**Describe the solution you'd like**
- [ ] #39185
- [x] #39186
- [x] #39187
- [x] #39188

What do you think @nextcloud/designers – any additional points, and who is interested in doing mockups for it? :)

**Comment by AndyScherzinger** [comment]
@jancborchardt since we also had this issue in the main view a while ago:

> Ability to search in the picker

* 🔎_Search_ or 🌪️_Filter_ (files currently does both, first filters current folder listing, yet searches everywhere in parallel)
* Where would you search? In the folder you are currently in? "Everywhere" files related? (bullet point (1) might implicitly answer it dep4ending on your answer to it)

**Comment by jancborchardt** [comment]
@AndyScherzinger we could start with Filter in current view rather than search. But also, out of the things listed, that is the least priority issue (also cause probably needs a lot of investment).

**Comment by nimishavijay** [comment]
Will pick this up for mockups! 

Additionally, it would be great to allow sharing via link from the file picker itself wherever appropriate. Related issue: https://github.com/nextcloud/mail/issues/6080 what do you think?

**Comment by jancborchardt** [comment]
> Additionally, it would be great to allow sharing via link from the file picker itself wherever appropriate. Related issue: https://github.com/nextcloud/mail/issues/6080 what do you think?

@nimishavijay sounds good as an additional optional mockup, yep! :)

**Comment by AndyScherzinger** [comment]
please align any specific @nimishavijay and @susnux - I guess sorting and remembering the last location are clear - however I would challenge the sorting part, but will do in the respective ticket.

**Comment by nimishavijay** [comment]
@susnux The new file picker can look like this:

![image](https://github.com/nextcloud/server/assets/52440189/9a8ee72b-b023-46a0-a0f8-7719fbe0f191)
![image](https://github.com/nextcloud/server/assets/52440189/c32495a1-8a72-4578-8711-91683d69b604)


- [ ] A similar layout to the settings modal is used to mimic the files app layout, with a navigation on the left containing:
  - [ ] All files
  - [ ] Recent 
  - [ ] Favorites
- [ ] The rest of the modal contains the content
  - [ ] If "All files" is chosen, the breadcrumbs are shown with the file list
  - [ ] If "Recent" is chosen, a heading "Recent items" is shown and the recent files list
  - [ ] If "Favorites" is chosen, a heading "Favorites" and the list of favorites
  - [ ] The new breadcrumbs component is used
  - [ ] Updated icons in file list: 20px [MDI folder icon](https://fonts.google.com/icons?selected=Material%20Icons%20Round%3Afolder%3A)
  - [ ] Remove the lines in between each item 
  - [ ] https://github.com/nextcloud/server/issues/39185
  - [ ] https://github.com/nextcloud/server/issues/39186
 

**Nice-to-haves**

- [ ] https://github.com/nextcloud/server/issues/39187 : the first item of the navigation can be a search bar which allows you to search for a file (across all your files) 

![image](https://github.com/nextcloud/server/assets/52440189/6d196e9f-587f-40cd-9b83-0e76cb2da48c)
![image](https://github.com/nextcloud/server/assets/52440189/33be2062-23b8-43a2-adaf-1a9af8a4fff0)

- [ ] https://github.com/nextcloud/mail/issues/6080 Wherever applicable, there could be the option to share a link from within the file picker. Currently, in Mail for eg, you have the option to choose that you want to share a link, and then the file picker opens, while this flow can be simplified by just having the option to share link form within the file picker anyway.

![image](https://github.com/nextcloud/server/assets/52440189/b2ea7edf-3252-4017-99db-7148f3244016)

**Comment by AndyScherzinger** [comment]
@nimishavijay @jancborchardt just since it got phrased differently in the latest mock up and description: Search should just be filtering for the moment and filtering would be on the current listing, no actual search on all your accessible files.

**Comment by nimishavijay** [comment]
@AndyScherzinger understood. then to indicate that search has been applied only on the current listing, the `Search results for "design"` heading can be under the breadcrumbs bar in the all files view and in the heading with a dot separator in recent and favorites (Recent items · Search results for "design"). cc @susnux and @nextcloud/designers

**Comment by marcoambrosini** [comment]
@nimishavijay I would strongly recommend to have the search-box on top of the search results like in some of your other mockups. I find it very counterintuitive that the search-box is filtering a list that it's not directly below it.

If the search-box was directly above the results, I think that there would be less need for a heading indicating that a filter is active, and it would be enough to have a `search in "design"` placeholder within the search input.

**Comment by susnux** [comment]
I like the design and it looks consistent :)
But for mobile I am not sure how this should look like. Should the left side menu simply be moved as a horizontal menu underneath the title (between title and current path)?

Because for the app settings, the left side simply is removed.

**Comment by nimishavijay** [comment]
> But for mobile I am not sure how this should look like. Should the left side menu simply be moved as a horizontal menu underneath the title (between title and current path)?

As discussed yesterday it can be a dropdown item which is shown above the breadcrumbs bar :)
